openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -157,301 +63,54 @@ def override_for_request(self, switch_name, active=True):
         """
         namespaced_switch_name = self._namespaced_name(switch_name)
         self._cached_switches[namespaced_switch_name] = active
-        log.info(u"%sSwitch '%s' set to %s for request.", self.log_prefix, namespaced_switch_name, active)
+        log.info(
+            "%sSwitch '%s' set to %s for request.",
+            self.log_prefix,
+            namespaced_switch_name,
+            active,
+        )
 
     @contextmanager
     def override_in_model(self, switch_name, active=True):
         """
         Overrides the active value for the given switch for the duration of this
         contextmanager.
         Note: The value is overridden in the model, not the request cache.
+        Note: This should probably be moved to a test class.
         """
         # Import is placed here to avoid model import at project startup.
+        # pylint: disable=import-outside-toplevel
         from waffle.testutils import override_switch as waffle_override_switch
+
         namespaced_switch_name = self._namespaced_name(switch_name)
         with waffle_override_switch(namespaced_switch_name, active):
-            log.info(u"%sSwitch '%s' set to %s in model.", self.log_prefix, namespaced_switch_name, active)
+            log.info(
+                "%sSwitch '%s' set to %s in model.",
+                self.log_prefix,
+                namespaced_switch_name,
+                active,
+            )
             yield
 
-    @property
-    def _cached_switches(self):
-        """
-        Returns a dictionary of all namespaced switches in the request cache.
-        """
-        return self._get_request_cache().setdefault('switches', {})
+
+class WaffleSwitch(BaseWaffleSwitch):
+    NAMESPACE_CLASS = WaffleSwitchNamespace
```

Comment:
> Ultimately, I'd like to not have WaffleSwitch and WaffleFlag exist any more in edx-platform. Do you agree, or think it is needed long term for some reason?

Yes, I agree. But those classes are needed to make use of the custom namespace classes. I'm not a big fan of adding a deprecation warning right now, because it's going to dump a lot of output in the console. I'd much rather make the switch (pun intended) in a separate PR after this is merged.

Reply:
:ok_hand: